pkg/linear-pools/contracts/silo-linear-pool/SiloLinearPool.test.ts

```diff
@@ -193,17 +199,29 @@ describe('SiloLinearPool', function () {
   describe('getWrappedTokenRate', () => {
     context('under normal operation', () => {
       it('returns the expected value', async () => {
-        // Calculate the expected rate and compare to the getWrappedToken return value
-        const assetStorage = await mockSilo.assetStorage(mainToken.address);
-        // Get the 4th member from the struct 'total deposits'
-        const totalAmount = assetStorage[3];
+        const expectedRate = fp(2);
+        expect(await pool.getWrappedTokenRate()).to.equal(expectedRate);
+      });
+    });
+
+    context('non-zero interest data', () => {
+      it('returns the expected value', async () => {
+        await mockInterestRateModel.setCompoundInterestRate(fp(1.5));
+        await mockInterestRateModel.setCurrentInterestRate(fp(1));
 
-        const totalShares: number = await wrappedToken.totalSupply();
+        await mockRepository.setProtocolShareFee(fp(0.01));
 
-        const expectedRate: number = totalAmount / totalShares;
+        await mockSilo.setInterestData(
+            mainToken.address, // interestBearingAsset
+            fp(1), // harvestedProtocolFees
+            fp(2), // protocolFees
+            0, // interestRateTimestamp
+            AssetStatus.Active // status
+        );
 
-        expect(await pool.getWrappedTokenRate()).to.equal(fp(expectedRate));
-      });
+        const expectedRate = fp(3.3365);
```

Comment:
For this one, I would just add a single-line comment saying "See details on Notion" or something like that.